### PR TITLE
Fixes `Humidifier:Steam:Gas` VariableInletWaterTemperature option

### PR DESCRIPTION
**Pull request by shorowit**
Pull request overview
---------------------
 - Fixes the issue reported at https://unmethours.com/question/98686/humidifiersteamgas-option-variableinletwatertemperature-not-working-energyplus/
 - The fix can be tested using [ExampleWithStorageandVarTemp.idf](https://github.com/NREL/EnergyPlus/files/14935213/ExampleWithStorageandVarTemp.idf.txt)

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

**Comment by Myoldmopar** [comment]
Everything is testing happily locally, if CI is good this can go right in.

File: src/EnergyPlus/Humidifiers.hh [hunk 1]
```diff
@@ -82,6 +82,8 @@ namespace Humidifiers {
         Variable,
         Num
     };
+    static constexpr std::array<std::string_view, static_cast<int>(InletWaterTemp::Num)> inletWaterTempsUC = {"FixedInletWaterTemperature",
+                                                                                                              "VariableInletWaterTemperature"};
```
Comment (rraustad) [inline src/EnergyPlus/Humidifiers.hh]: inletWaterTempsUC? doesn't look UC.

Comment (Myoldmopar) [reply src/EnergyPlus/Humidifiers.hh]: > inletWaterTempsUC? doesn't look UC.

Was I supposed to actually fix them or something?  


















Thanks @rraustad :)

**Comment by Myoldmopar** [comment]
Good now.  :)  Thanks again @rraustad.  Merging.

**Comment by Myoldmopar** [comment]
And thanks for the initial fix @shorowit !

**Pull request merged**